lib/nodejs/lib/thrift/thrift.js

```diff
@@ -219,6 +222,21 @@ copyMap = function(obj, types){
   }
   var Type = type;
 
+  if (obj instanceof Map) {
```

Comment:
what is happening on line 226? and 231?

i understand that you're following the example on lines 240-256; am wondering if want to use clearer JS syntax (and possibly es6?)